ofono/plugins/ril.c

```diff
@@ -475,6 +480,46 @@ void ril_switchUser()
 
 }
 
+/* TODO: Reading RILD socket path by for now from rilmodem .conf file,
+ * 	but change this later to StateFs when plans are more concrete.
+ * return: Null-terminated path string. Ownership transferred.
+ * */
+static char *ril_socket_path()
+{
+	GError *err = NULL;
+	GKeyFile *keyfile = NULL;
+	char *res = NULL;
+
+	keyfile = g_key_file_new();
+	g_key_file_set_list_separator(keyfile, ',');
+
+	if (!g_key_file_load_from_file(keyfile, RILMODEM_CONF_FILE, 0, &err)) {
+		if (err) {
+			DBG("conf load result: %s", err->message);
+			g_error_free(err);
+		}
+	} else {
+		if (g_key_file_has_group(keyfile, RILSOCK_CONF_GROUP)) {
+			res = g_key_file_get_string(
+				keyfile, RILSOCK_CONF_GROUP, RILSOCK_CONF_PATH, &err);
+			if (err) {
+				DBG("conf get result: %s", err->message);
+				g_error_free(err);
+			}
+		}
+	}
+
+	if (keyfile)
+		g_key_file_free(keyfile);
```

Comment:
Check for NULL is unnecessary because `g_key_file_new()` never returns NULL:

```C
GKeyFile *
g_key_file_new (void)
{
  GKeyFile *key_file;

  key_file = g_slice_new0 (GKeyFile);
  key_file->ref_count = 1;
  g_key_file_init (key_file);

  return key_file;
}
```